security/advancedtls/advancedtls.go

```diff
@@ -184,6 +197,9 @@ type ClientOptions struct {
 	// RevocationConfig is the configurations for certificate revocation checks.
 	// It could be nil if such checks are not needed.
 	RevocationConfig *RevocationConfig
+	// VersionOptions contains the requirements for the TLS version being used.
+	// If not set, the default values would be used.
+	VersionOptions TLSVersionOptions
```

Comment:
I'd probably remove this extra type and put `MinVersion` and `MaxVersion` directly in `Client/ServerOptions`.  It seems to save nothing this way, since `TLSVersionOptions` isn't shared, and usage is more difficult for users, as it requires cross-referencing different structs, and setting via nested structs.

Reply:
Friendly ping on this comment; I think we're still waiting for an update here.